File: backend/individual-assignment/typescript/src/product.ts
export enum ProductType {
  PHYSICAL = "PHYSICAL",
  BOOK = "BOOK",
  DIGITAL = "DIGITAL",
  MEMBERSHIP = "MEMBERSHIP",
}

export class Product {
  constructor(
    public name: string,
    public type: ProductType,
    public price: number,
  ) {}
}

File: backend/shopping/src/model/product.py
from enum import Enum

from attr import dataclass


class ProductType(Enum):
    PHYSICAL = 'physical'
    BOOK = 'book'
    DIGITAL = 'digital'
    MEMBERSHIP = 'membership'


@dataclass(frozen=True)
class Product:
    name: str
    price: float
    product_type: ProductType
